test/contrib/http/request_test.rb

```diff
@@ -0,0 +1,112 @@
+require 'time'
+require 'contrib/elasticsearch/test_helper'
+require 'ddtrace'
+require 'helper'
+require 'json'
+
+class HTTPRequestTest < Minitest::Test
+  ELASTICSEARCH_HOST = '127.0.0.1'.freeze
+  ELASTICSEARCH_PORT = 49200
+  ELASTICSEARCH_SERVER = ('http://' +
+                          HTTPIntegrationTest::ELASTICSEARCH_HOST + ':' +
+                          HTTPIntegrationTest::ELASTICSEARCH_PORT.to_s).freeze
+
+  def setup
+    @tracer = get_test_tracer
+
+    # wait until it's really running, docker-compose can be slow
+    wait_http_server 'http://' + ELASTICSEARCH_HOST + ':' + ELASTICSEARCH_PORT.to_s, 60
+
+    @client = Net::HTTP.new(ELASTICSEARCH_HOST, ELASTICSEARCH_PORT)
+    pin = Datadog::Pin.get_from(@client)
+    pin.tracer = @tracer
+  end
+
+  def test_get_request
+    response = @client.get('_cluster/health')
+    assert_equal('200', response.code, 'bad response status')
+    content = JSON.parse(response.body)
+    assert_kind_of(Hash, content, 'bad content')
+    spans = @tracer.writer.spans()
+    assert_equal(1, spans.length)
```

Comment:
Yes in the case of ``rails-cache`` it's quite different because when you're doing cache operations you may have multiple calls to your cache backend (or multiple backends) for just one ``Rails.cache.fetch()`` (it depends by the app and by the underlying caching mechanism). Furthermore, if you use directly the Redis client, it behaves like it should be.

In this case we're always having 2 spans and the fact that we're instrumenting the HTTP client is not related to give more details to the ES call, since it's just an implementation detail that it uses HTTP. We should write something on docs, but bear in mind that we may need to change this behavior in the future.